org.eclipse.draw2d/src/org/eclipse/draw2d/graph/RankAssignmentSolver.java

```diff
@@ -273,32 +267,34 @@ void updateSubgraph(Node root) {
 		}
 	}
 
+	@Override
 	public void visit(DirectedGraph graph) {
 		this.graph = graph;
 		initCutValues();
 		networkSimplexLoop();
-		if (graph.forestRoot == null)
+		if (graph.forestRoot == null) {
 			graph.nodes.normalizeRanks();
-		else
+		} else {
 			normalizeForest();
+		}
 	}
 
 	private void normalizeForest() {
 		NodeList tree = new NodeList();
 		graph.nodes.resetFlags();
 		graph.forestRoot.flag = true;
 		EdgeList rootEdges = graph.forestRoot.outgoing;
-		Stack stack = new Stack();
-		for (int i = 0; i < rootEdges.size(); i++) {
-			Node node = rootEdges.getEdge(i).target;
+		Deque<Node> stack = new ArrayDeque<>();
+		for (Edge e : rootEdges) {
+			Node node = e.target;
 			node.flag = true;
 			stack.push(node);
 			while (!stack.isEmpty()) {
-				node = (Node) stack.pop();
+				node = stack.pop();
 				tree.add(node);
-				Iterator neighbors = node.iteratorNeighbors();
+				Iterator<Node> neighbors = node.iteratorNeighbors();
 				while (neighbors.hasNext()) {
-					Node neighbor = (Node) neighbors.next();
+					Node neighbor = neighbors.next();
```

Comment:
You can probably get rid of the iterator completely via `node.iteratorNeighbors().forEachRemaining(neighbor -> {...}`

Reply:
This should be part of a separate pull request. Will fix it there.